Admin/Admin.php

```diff
@@ -2116,49 +2107,113 @@ public function buildBreadcrumbs($action, MenuItemInterface $menu = null)
 
-            $menu = $menu->addChild(
+            $menu->addChild(
                 $this->trans($this->getLabelTranslatorStrategy()->getLabel('dashboard', 'breadcrumb', 'link'), array(), 'SonataAdminBundle'),
                 array('uri' => $this->routeGenerator->generate('sonata_admin_dashboard'))
             );
         }
 
-        $menu = $menu->addChild(
+        $childMenu = $menu->addChild(
             $this->trans($this->getLabelTranslatorStrategy()->getLabel(sprintf('%s_list', $this->getClassnameLabel()), 'breadcrumb', 'link')),
             array('uri' => $this->hasRoute('list') && $this->isGranted('LIST') ? $this->generateUrl('list') : null)
         );
 
+        $this->addBreadcrumbsDropdown($childMenu);
+
         $childAdmin = $this->getCurrentChildAdmin();
 
         if ($childAdmin) {
             $id = $this->request->get($this->getIdParameter());
 
-            $menu = $menu->addChild(
+            $childMenu = $menu->addChild(
                 $this->toString($this->getSubject()),
                 array('uri' => $this->hasRoute('edit') && $this->isGranted('EDIT') ? $this->generateUrl('edit', array('id' => $id)) : null)
             );
 
+            $this->addBreadcrumbsDropdown($childMenu, $this->getSubject());
+            $this->configureTabMenu($childMenu, 'action', $childAdmin);
+
             return $childAdmin->buildBreadcrumbs($action, $menu);
         } elseif ($this->isChild()) {
             if ($action == 'list') {
                 $menu->setUri(false);
             } elseif ($action != 'create' && $this->hasSubject()) {
-                $menu = $menu->addChild($this->toString($this->getSubject()));
+                $childMenu = $menu->addChild($this->toString($this->getSubject()));
+
+                $this->addBreadcrumbsDropdown($childMenu, $this->getSubject());
             } else {
-                $menu = $menu->addChild(
+                $menu->addChild(
                     $this->trans($this->getLabelTranslatorStrategy()->getLabel(sprintf('%s_%s', $this->getClassnameLabel(), $action), 'breadcrumb', 'link'))
                 );
             }
         } elseif ($action != 'list' && $this->hasSubject()) {
-            $menu = $menu->addChild($this->toString($this->getSubject()));
+            $menu->addChild($this->toString($this->getSubject()));
         } elseif ($action != 'list') {
-            $menu = $menu->addChild(
+            $childMenu = $menu->addChild(
                 $this->trans($this->getLabelTranslatorStrategy()->getLabel(sprintf('%s_%s', $this->getClassnameLabel(), $action), 'breadcrumb', 'link'))
             );
+
+            $this->addBreadcrumbsDropdown($childMenu);
         }
 
         return $this->breadcrumbs[$action] = $menu;
     }
 
+    private function addBreadcrumbsDropdown(MenuItemInterface $menu, $subject = null)
+    {
+        if ($subject) {
+            $id = $this->request->get($this->getIdParameter());
+
+            $links = array(
```

Comment:
these links are already defined anywhere... we should use them from there, so we don't get different action dropdowns